pkg/prowgen/prowgen_test.go

```diff
@@ -349,7 +371,9 @@ func TestGenerateJobs(t *testing.T) {
 	for _, tc := range tests {
 		t.Run(tc.id, func(t *testing.T) {
 			jobConfig := GenerateJobs(tc.config, tc.repoInfo, jobconfig.Generated)
-			pruneForTests(jobConfig) // prune the fields that are tested in TestGeneratePre/PostsubmitForTest
+			if !tc.keep {
```

Comment:
This smells. IIRC this means a lower-level test handles those specifics.

Reply:
The higher level test doesn’t test that pod specs are valid.  So neither unit tests the influence of pod spec.